File: preschool-management-backend/src/main/java/com/vusystem/preschool_management_backend/modules/mobile/service/LeaveRequestService.java
package com.vusystem.preschool_management_backend.modules.mobile.service;

import com.vusystem.preschool_management_backend.modules.mobile.dto.request.LeaveRequestCreateRequest;
import com.vusystem.preschool_management_backend.modules.mobile.dto.response.LeaveRequestResponse;

import java.util.List;

public interface LeaveRequestService {
    LeaveRequestResponse createRequest(LeaveRequestCreateRequest request);
    List<LeaveRequestResponse> getParentRequests(Long childId);
    void updateStatus(Long id, String status); // thiết kế dùng chung cho teacher và admin mở rộng sau này
    List<LeaveRequestResponse> getClassRequests(Long classId);
}

File: mobile/src/services/leaveRequest.service.ts

import axiosInstance from '../config/api.client';

export interface LeaveRequestCreateRequest {
  childId: number;
  startDate: string;
  endDate: string;
  reason: string;
}

export interface LeaveRequestResponse {
  id: number;
  childId: number;
  childFullName: string;
  startDate: string;
  endDate: string;
  reason: string;
  status: 'PENDING' | 'APPROVED' | 'REJECTED';
}

export const leaveRequestService = {
  createRequest: async (data: LeaveRequestCreateRequest): Promise<LeaveRequestResponse> => {
    const response = await axiosInstance.post<{ status: number; message: string; data: LeaveRequestResponse }>(
      '/mobile/leave-requests',
      data
    );
    return response.data.data;
  },

  getParentRequests: async (childId: number): Promise<LeaveRequestResponse[]> => {
    const response = await axiosInstance.get<{ status: number; message: string; data: LeaveRequestResponse[] }>(
      `/mobile/leave-requests/children/${childId}`
    );
    return response.data.data;
  },

  getClassRequests: async (classId: number): Promise<LeaveRequestResponse[]> => {
    const response = await axiosInstance.get<{ status: number; message: string; data: LeaveRequestResponse[] }>(
      `/mobile/leave-requests/classes/${classId}`
    );
    return response.data.data;
  },

  updateStatus: async (id: number, status: 'APPROVED' | 'REJECTED'): Promise<void> => {
    await axiosInstance.put(
      `/mobile/leave-requests/${id}/status`,
      null,
      { params: { status } }
    );
  }
};
